controllers/inverse_kinematics_pyikfast/inverse_kinematics_pyikfast.py

```diff
@@ -80,27 +80,74 @@
             sensors.append(sensor)
             sensor.enable(timeStep)
         except Exception as e:
-            print('Rotational Motor: ' + device.getName() +
-                  ' has no Position Sensor')
+            print("Rotational Motor: " + device.getName() + " has no Position Sensor")
+print("minPositions", minPositions)
+print("maxPositions", maxPositions)
+
 # --------------------------------------------------------------------
+# move the arm onto target using known solution
+target_position_joint_state = np.array([0, -np.pi / 2, np.pi / 2, 0, 0, 0]).astype(
+    np.double
+)
+def move_to_joint_state(motors, joint_state):
+    """Move robot endpoint to a known position that aligns it perfectly with the target"""
+    for i in range(len(joint_state)):
+        motors[i].setPosition(joint_state[i])
+# plase end-effector on target
+move_to_joint_state(motors, target_position_joint_state)
 
+# --------------------------------------------------------------------
+# show that the endpoint pose is indeed the targtet pose
 # Initialize our inverse kinematics module
 ik = inverseKinematics(pyikfast_module, protoName, minPositions, maxPositions)
+print("ik", ik)
+
+fk_pos, fk_rot = ik.get_fk(target_position_joint_state)
+print("fk_pos", fk_pos)
+print("fk_rot", fk_rot)
+
+# check that the target position and rotation matches te end-effector's
+target = supervisor.getFromDef("TARGET")
+# target position relative to world.
+target_pos_world = np.array(target.getPosition())
+# target rotation relative to world
+target_rot_world = np.array(target.getOrientation()).reshape(3, 3)
+assert np.allclose(fk_pos, target_pos_world)
+assert np.allclose(fk_rot, target_rot_world)
+print("Yay, target pose and end-effector pose match")
+
+# --------------------------------------------------------------------
+# Try is get an IK solution for the current end-point pose.
+# there should be at least one solution, target_position_joint_state
+ikResults = ik.get_ik(fk_pos, fk_rot)
+
+assert len(ikResults) > 0
+
+#############################################################################################
+# We don't get any further
+print("Yay, We found in IK solution to our current end effector pose")
+
 
 # Initialize the RelativePositions module. 'TARGET' is the DEF of the spawned sphere.
 # You can change this DEF to any other object. You can also initialilze several like this:
 # RelPos_1 = RelativePositions(supervisor, 'TARGET1')
 # RelPos_2 = RelativePositions(supervisor, 'TARGET2')
-RelPos = RelativePositions(supervisor, 'TARGET')
-
+RelPos = RelativePositions(supervisor, "TARGET")
+print("RelPos.get_pos()", RelPos.get_pos())
```

Comment:
Why change to double quote? Also this extra output might be a bit too verbose. Add it to troubleshooting-controller